PROTO_tests/tests/groupby_test.py

```diff
@@ -0,0 +1,577 @@
+
+    def make_arrays(self, size):
+        keys = np.random.randint(0, self.GROUPS, size, dtype=np.uint64)
+        keys2 = np.random.randint(0, self.GROUPS, size)
+        i = np.random.randint(0, size // self.GROUPS, size)
+        u = np.random.randint(0, size // self.GROUPS, size, dtype=np.uint64)
+        f = np.random.randn(size)  # normally dist random numbers
+        b = (i % 2) == 0
+        d = {"keys": keys, "keys2": keys2, "int64": i, "uint64": u, "float64": f, "bool": b}
+
+        return d
+
+    def groupby_to_arrays(self, df: pd.DataFrame, kname, vname, op, levels):
+        g = df.groupby(kname)[vname]
+        agg = g.aggregate(op.replace("arg", "idx"))
+        if op == "prod":
+            # There appears to be a bug in pandas where it sometimes
+            # reports the product of a segment as NaN when it should be 0
+            agg[agg.isna()] = 0
+        if levels == 1:
+            keys = agg.index.values
+        else:
+            keys = tuple(zip(*(agg.index.values)))
+        return keys, agg.values
+
+    def compare_keys(self, pdkeys, akkeys, levels, pdvals, akvals):
+        """
+        Compares the numpy and arkouda arrays via the numpy.allclose method with the
+        default relative and absolute tolerances, returning 0 if the arrays are similar
+        element-wise within the tolerances, 1 if they are dissimilar.element
+
+        :return: 0 (identical) or 1 (dissimilar)
+        :rtype: int
+        """
+        if levels == 1:
+            assert np.allclose(pdkeys, akkeys.to_ndarray())  # key validation
+        else:
+            for lvl in range(levels):
+                assert np.allclose(pdkeys[lvl], akkeys[lvl].to_ndarray())
+
+        assert np.allclose(pdvals, akvals.to_ndarray(), equal_nan=True)  # value validation
+
+    @pytest.mark.parametrize("size", pytest.prob_size)
+    @pytest.mark.parametrize("levels", LEVELS)
+    def test_pandas_equivalency(self, size, levels):
+        data = self.make_arrays(size)
+        df = pd.DataFrame(data)
+        akdf = {k: ak.array(v) for k, v in data.items()}
+        if levels == 1:
+            akg = ak.GroupBy(akdf["keys"])
+            keyname = "keys"
+        elif levels == 2:
+            akg = ak.GroupBy([akdf["keys"], akdf["keys2"]])
+            keyname = ["keys", "keys2"]
+
+        pdkeys, pdvals = self.groupby_to_arrays(df, keyname, "int64", "count", levels)
+        akkeys, akvals = akg.count()
+
+        self.compare_keys(pdkeys, akkeys, levels, pdvals, akvals)
+
+        for vname in ("int64", "uint64", "float64", "bool"):
+            for op in ak.GroupBy.Reductions:
+                do_check = True
+                try:
+                    pdkeys, pdvals = self.groupby_to_arrays(df, keyname, vname, op, levels)
+                except Exception:
+                    do_check = False
+                try:
+                    akkeys, akvals = akg.aggregate(akdf[vname], op)
+                except Exception as E:
```

Comment:
After discussing with Pierce. We decided it would be best to leave the prints always on for now and they can be used by devs if needed. Otherwise, they are not harming anything.